app/services/auth_methods_session.rb

```diff
@@ -0,0 +1,16 @@
+class AuthMethodsSession
+  attr_reader :user_session
+
+  def initialize(user_session:)
+    @user_session = user_session
+  end
+
+  def authenticate!(auth_method)
+    user_session[TwoFactorAuthenticatable::NEED_AUTHENTICATION] = false
+    auth_events << { auth_method:, at: Time.zone.now }
```

Comment:
Is it worth limiting to the last e.g. 10 authentication attempts?

Reply:
That's a good thought. I think a limit would be reasonable. I can add that.

Reply:
Updated in 3a0984243